tests/integration/targets/aci_management_network_instance_profile/tasks/main.yml

```
@@ -0,0 +1,226 @@
+    subnets:
+      - 10.20.30.0/24
+      - 192.168.11.0/24
+    state: present
+  check_mode: false
+  register: update_profile
+
+- name: Verify object update
+  ansible.builtin.assert:
+    that:
+      - update_profile is changed
+      - update_profile.previous == nm_add_profile_again.current
+      - update_profile.current.0.mgmtInstP.attributes.name == "ans_network_inst_profile"
+      - update_profile.current.0.mgmtInstP.attributes.prio == "level1"
+      - update_profile.current.0.mgmtInstP.attributes.descr == "Updated Ansible test profile"
+      - update_profile.current.0.mgmtInstP.children.0.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - update_profile.current.0.mgmtInstP.children.1.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - update_profile.current.0.mgmtInstP.children | length == 2
+      - update_profile.proposed.mgmtInstP.attributes.name == "ans_network_inst_profile"
+      - update_profile.proposed.mgmtInstP.attributes.prio == "level1"
+      - update_profile.proposed.mgmtInstP.attributes.descr == "Updated Ansible test profile"
+      - update_profile.proposed.mgmtInstP.children.0.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - update_profile.proposed.mgmtInstP.children.1.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - update_profile.proposed.mgmtInstP.children.2.mgmtSubnet.attributes.ip == "192.168.10.0/24"
+      - update_profile.proposed.mgmtInstP.children.2.mgmtSubnet.attributes.status == "deleted"
+
+# QUERY PROFILE
+- name: Query Management Network Instance Profile
+  cisco.aci.aci_management_network_instance_profile:
+    <<: *aci_info
+    name: ans_network_inst_profile
+    state: query
+  register: query_profile
+
+- name: Create Second Management Network Instance Profile
+  cisco.aci.aci_management_network_instance_profile:
+    <<: *aci_info
+    name: ans_second_network_inst_profile
+    state: present
+
+- name: Query All Management Network Instance Profiles
+  cisco.aci.aci_management_network_instance_profile:
+    <<: *aci_info
+    state: query
+  register: query_all_profiles
+
+- name: Verify object query
+  ansible.builtin.assert:
+    that:
+      - query_profile is not changed
+      - query_profile.current.0.mgmtInstP.attributes.name == "ans_network_inst_profile"
+      - query_profile.current.0.mgmtInstP.attributes.prio == "level1"
+      - query_profile.current.0.mgmtInstP.attributes.descr == "Updated Ansible test profile"
+      - query_profile.current.0.mgmtInstP.children.0.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - query_profile.current.0.mgmtInstP.children.1.mgmtSubnet.attributes.ip in ["10.20.30.0/24", "192.168.11.0/24"]
+      - query_profile.current.0.mgmtInstP.children | length == 2
+      - query_profile.proposed == {}
+      - query_all_profiles is not changed
+      - query_all_profiles.proposed == {}
+      - query_all_profiles.current | length == 2
```

```suggestion
      - query_all_profiles.current | length >= 2
```
This is because the module returns all `mgmtInstP` objects in the result.